src/components/SchemaRow/TopLevelSchemaRow.tsx

```diff
@@ -1,85 +1,107 @@
 import { faCaretDown } from '@fortawesome/free-solid-svg-icons/faCaretDown.js';
 import { isRegularNode, RegularNode } from '@stoplight/json-schema-tree';
-import { Box, Flex, Icon, Menu, Pressable } from '@stoplight/mosaic';
+import { Box, Flex, HStack, Icon, Menu, Pressable } from '@stoplight/mosaic';
+import { useUpdateAtom } from 'jotai/utils';
 import * as React from 'react';
 
-import { NEGATIVE_NESTING_OFFSET } from '../../consts';
+import { COMBINER_NAME_MAP } from '../../consts';
+import { useIsOnScreen } from '../../hooks/useIsOnScreen';
 import { calculateChildrenToShow, isComplexArray } from '../../tree';
+import { showPathCrumbsAtom } from '../PathCrumbs';
 import { ChildStack } from '../shared/ChildStack';
 import { SchemaRow, SchemaRowProps } from './SchemaRow';
 import { useChoices } from './useChoices';
 
-export const TopLevelSchemaRow: React.FC<SchemaRowProps> = ({ schemaNode, nestingLevel }) => {
+export const TopLevelSchemaRow = ({ schemaNode }: Pick<SchemaRowProps, 'schemaNode'>) => {
```

Comment:
is there any reason we dropped `React.FC`? It also ensures the return type is correct, while this one doesn't do it anymore which can potentially lead to invalid renders if undefined is returned.

Reply:
Yeah `React.FC` is generally over-used in a few places. It adds props to the typings such as `children` that are not actually supported or used by this component.

Re the return undefined thing - if you try adding a `return undefined` anywhere in here you'll see that the typings call it out wherever this component is used (such as the `JsonSchemaViewer` component).

Reply:
> Yeah React.FC is generally over-used in a few places. It adds props to the typings such as children that are not actually supported or used by this component.

because `children` is something that's always supported? even if not consumed, you can still pass it around, it's just some sort of a native property supported by React :man_shrugging: 
I dunno, this feels just less correct than having `React.FC`, but it's a topic for another convo.

Reply:
This component does not do anything with children - the consumer should not pass in children or think that they can and it would do something. It's like adding a random typing to your component that is not used at all